File: src/main/java/net/ndrei/teslapoweredthingies/machines/ItemLiquefierBlock.java
package net.ndrei.teslapoweredthingies.machines;

import net.minecraft.init.Items;
import net.minecraft.item.ItemStack;
import net.minecraft.item.crafting.IRecipe;
import net.minecraftforge.oredict.ShapedOreRecipe;
import net.ndrei.teslacorelib.TeslaCoreLib;

/**
 * Created by CF on 2017-04-09.
 */
public class ItemLiquefierBlock extends BaseThingyBlock<FluidSolidifierEntity> {
    public ItemLiquefierBlock() {
        super("item_liquefier", ItemLiquefierEntity.class);
    }

    @Override
    protected IRecipe getRecipe() {
        return new ShapedOreRecipe(new ItemStack(this, 1),
                "bob", "oco", "ogo",
                'b', Items.LAVA_BUCKET,
                'c', TeslaCoreLib.machineCase,
                'o', "obsidian", // Blocks.OBSIDIAN,
                'g', "gearIron" // TeslaCoreLib.gearIron
        );
    }
}

File: src/main/kotlin/net/ndrei/teslapoweredthingies/machines/itemliquefier/ItemLiquefierBlock.kt
package net.ndrei.teslapoweredthingies.machines.itemliquefier

import net.minecraft.init.Items
import net.minecraft.item.ItemStack
import net.minecraft.item.crafting.IRecipe
import net.minecraftforge.oredict.ShapedOreRecipe
import net.ndrei.teslacorelib.GearRegistry
import net.ndrei.teslacorelib.MATERIAL_IRON
import net.ndrei.teslacorelib.annotations.AutoRegisterBlock
import net.ndrei.teslacorelib.items.MachineCaseItem
import net.ndrei.teslapoweredthingies.machines.BaseThingyBlock

/**
 * Created by CF on 2017-06-30.
 */
@AutoRegisterBlock
object ItemLiquefierBlock
    : BaseThingyBlock<ItemLiquefierEntity>("item_liquefier", ItemLiquefierEntity::class.java) {

    override val recipe: IRecipe?
        get() = ShapedOreRecipe(null, ItemStack(this, 1),
                "bob", "oco", "ogo",
                'b', Items.LAVA_BUCKET,
                'c', MachineCaseItem,
                'o', "obsidian",
                'g', GearRegistry.getMaterial(MATERIAL_IRON)?.oreDictName ?: "gearIron"
        )
}